File: src/main/java/ro/anproca/example/PythonTest.java
package ro.anproca.example;

import org.apache.thrift.TException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.transport.TFramedTransport;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;
import tutorial.NotificationService;

/**
 * Created by alex on 12.12.2014.
 */
public class PythonTest {


    public static void main(String[] args) throws TException {

        TTransport transport = new TFramedTransport(new TSocket("localhost", 12005));

        TProtocol protocol = new TBinaryProtocol(transport);

        NotificationService.Client callbackClient = new NotificationService.Client(protocol);

        transport.open();
        callbackClient.notify("Hello from java");
        transport.close();

    }

}

File: src/main/python/pythonTest.py
#!/usr/bin/env python
__author__ = 'alex'

import sys
sys.path.append('../../../target/generated-sources/thrift/py')

from tutorial import MultiplicationService
from tutorial import NotificationService
from tutorial.ttypes import *

from thrift import Thrift
from thrift.transport import TSocket, TTransport
from thrift.protocol import TBinaryProtocol
from thrift.server import TServer, TNonblockingServer
from multiprocessing import Process
from time import sleep

try:
    # Make socket
    transport = TSocket.TSocket('localhost', 12005)

    # Buffering is critical. Raw sockets are very slow
    transport = TTransport.TBufferedTransport(transport)

    # Wrap in a protocol
    protocol = TBinaryProtocol.TBinaryProtocol(transport)

    # Create a client to use the protocol encoder
    client = NotificationService.Client(protocol)

    # Connect!
    transport.open()

    msg = client.notify("Hello from python")

    transport.close()

except Thrift.TException, tx:
    print "%s" % (tx.message)